osu.Game/Screens/Select/SongSelect.cs

```diff
@@ -311,6 +311,18 @@ private void performUpdateSelected()
 
             void run()
             {
+                if (Beatmap.Disabled)
```

Comment:
i really dislike this logic. can this nor just check if `Beatmap.Disabled` and not change selection?

Reply:
It's scheduled with a delay so we can't check earlier. The display has already changed in the carousel if it was clicked but the actual active beatmap hasn't undated yet so entering a level will enter one different than the one indicated.

Reply:
There has to be a better way to do this. Adding code like this is going in the wrong direction – it's piling more hacks on top of sub-optimal existing code.